### PR TITLE
Add initial version of Jenkinsfile source sync

### PR DESCRIPTION
**Pull request by JPinkney**
### What does this PR do?
This PR adds a Jenkinsfile for syncing the sources between upstream and downstream. Includes updating the base image when syncing to downstream

### What issues does this PR fix or reference?
https://issues.redhat.com/browse/WTO-23

### Is it tested? How?

Tested on https://codeready-workspaces-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/web-terminal-sync-web-terminal-exec/

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+def SOURCE_REPO = "redhat-developer/web-terminal-exec" //source repo from which to find and sync commits to pkgs.devel repo
+def DWNSTM_REPO = "containers/web-terminal-exec" // dist-git repo to use as target for everything
+
+def SOURCE_BRANCH = "v1.0.x"         // target branch in GH repo, eg., master or v1.0.x
+def DWNSTM_BRANCH = "web-terminal-1.0-rhel-8" // target branch in dist-git repo, eg., web-terminal-1.0-rhel-8
+
+def buildNode = "rhel7-releng" // slave label
+timeout(120) {
+	node("${buildNode}"){ stage "Sync repos"
```
Comment (amisevsk) [inline Jenkinsfile:11]:
```suggestion
    node("${buildNode}"){ stage "Sync repos"
```

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+            export SOURCE_BRANCH=''' + SOURCE_BRANCH + '''
+            export DWNSTM_REPO=''' + DWNSTM_REPO + '''
+            export DWNSTM_BRANCH=''' + DWNSTM_BRANCH + '''
+            export CRW_KEYTAB=''' + CRW_KEYTAB + '''
+            export USER=''' + USER + '''
+            curl -L -s -S https://raw.githubusercontent.com/redhat-developer/web-terminal-operator/master/bootstrap-sync.sh -o ./sync.sh
+            chmod +x ./sync.sh
+            ./sync.sh
+
+            # initialize kerberos
```
Comment (amisevsk) [inline Jenkinsfile:39]: Doens't `bootstrap-sync.sh` initialize kerberos? Not that it really matters if we do it twice.

**Review by amisevsk — APPROVED** [review]
LGTM, though I won't pretend to be an expert on the subject matter.

File: Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+            klist # verify working
+
+            cd "${WORKSPACE}/sources"
+            SOURCE_SHA=$(git rev-parse HEAD)
+            cd ..
+
+            cd "${WORKSPACE}/targetdwn"
+
+            # Update the base image
+            curl -L -s -S https://raw.githubusercontent.com/redhat-developer/codeready-workspaces/master/product/updateBaseImages.sh -o /tmp/updateBaseImages.sh
```
Comment (sleshchenko) [inline Jenkinsfile:51]: I just wonder if it makes a bit more sense to update base image in the upstream and then these changes would be just synced to downstream. I'm OK with leaving as is as well or leaving that here to double-check that the latest image is used as well.

File: Jenkinsfile [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+              git commit -s -m "[sync] Updated from ${SOURCE_REPO} @ ${SOURCE_SHA:0:8} "
+              git push origin "${DWNSTM_BRANCH}"
+              echo "[sync] Updated from ${SOURCE_REPO} @ ${SOURCE_SHA:0:8} "
+            else
+              echo "Source and downstream contents are the same. No need to sync"
+            fi
+            '''
+      }
+    }
+	}
```
Comment (sleshchenko) [inline Jenkinsfile:66]: It does not seem to be well-formatted. Try https://codebeautify.org/javaviewer or any another tool to auto-format that

**Review by sleshchenko — APPROVED** [review]

File: Jenkinsfile (hunk 2 above)
Comment (JPinkney) [reply Jenkinsfile]: Yeah, I'll have to remove it from there. For some reason, it doesn't initialize correctly when using the script but works perfectly fine when its run inside of the jenkinsfile :shrug:

**Pull request merged**